// This file listes the protobuffer messages used in the example to make a simple uart communcation protocol. 
// Commands are send from the desktop to the microctontroller which in turn sends a reply. 

syntax = "proto3";
 
import "shared/lora_phy.proto";

option csharp_namespace = "LoRa";

message TransmitCommand  
{ 
    bool IsMulticast = 1; 
    uint32 DeviceId = 2; 
    uint32 SequenceNumber = 3;
    // Period <50 will disable it
    optional uint32 Period = 4;
    optional uint32 MaxPacketCount = 5;
    bytes Payload = 6;
}
 
message RequestBootInfo { 
    bool Request = 1; 
} 
 
message DeviceConfiguration {
    bool EnableAlwaysSend = 1;
    uint32 AlwaysSendPeriod = 2;
}

message ClearMeasurementsCommand {
    // No content to report
    bool SendBootAfter = 1;
}

message UartCommand { 
    oneof Body { 
        RadioRxConfig rxConfig = 1; 
        RadioTxConfig txConfig = 2; 
        DeviceConfiguration deviceConfiguration = 3;
        TransmitCommand transmitCommand = 4;
        RequestBootInfo requestBootInfo = 5; 
        ClearMeasurementsCommand clearMeasurementsCommand = 6;
    } 
} 
